ex2_system/source/beacon/beacon_task.c

```diff
@@ -32,13 +32,33 @@ SAT_returnState start_beacon_daemon(void);
  * @param pvParameters
  *    task parameters (not used)
  */
-static void * beacon_daemon(void *pvParameters) {
-    TickType_t delay = pdMS_TO_TICKS(1000);
-    for ( ;; ) {
-        // TODO construct and send the system beacon
+static void *beacon_daemon(void *pvParameters) {
+  TickType_t delay = pdMS_TO_TICKS(1000);  // make this config.able
+  for (;;) {
+    int8_t uhf_status = -1;
+    /* Constructing the system beacon content */
+    // Refer to table 3 of MOP
+    UHF_configStruct beacon_msg;
+    // TODO: call the appropriate HAL functions to get the most updated or
+    // cached information of the components + state machine, RTC, etc.

```

Comment:
Probably sth we'll come back to in a few months. Not certain information right now.